java/ql/lib/semmle/code/java/frameworks/Lastaflute.qll

```diff
@@ -0,0 +1,26 @@
+/**
+ * Provides classes and predicates for working with the Lastaflute web framework.
+ */
+
+import java
+import semmle.code.java.dataflow.FlowSources
+
+/**
+ * The `org.lastaflute.web.Execute` annotation.
+ */
+class LastafluteExecuteAnnotation extends Annotation {
+  LastafluteExecuteAnnotation() { this.getType().hasQualifiedName("org.lastaflute.web", "Execute") }
+}
+
+/**
+ * The parameter of a method defining a URL handler using the Lastaflute framework.
+ */
+class LastafluteHandlerParameterSource extends RemoteFlowSource {
+  LastafluteHandlerParameterSource() {
+    exists(Parameter p | p.getCallable().getAnAnnotation() instanceof LastafluteExecuteAnnotation |
+      p = this.asParameter()
+    )
```

Comment:
I suggest writing this using MaD:
```
- ["org.lastaflute.web", "Execute", False, "", "", "Annotated", "Parameter", "remote", "manual"]
```

Reply:
See e.g. the qltests in `java/ql/test/library-tests/dataflow/external-models/` to see what's possible using the `"Annotated"` functionality of MaD.